common/metrics/interfaces.go

```diff
@@ -104,7 +104,7 @@ type (
 		RecordTimer(timer string, d time.Duration)
 		// RecordDistribution records a distribution (wrapper on top of timer) for the given
 		// metric name
-		RecordDistribution(id string, d int)
```

Comment:
we really need to consolidate userScope with internal metric client.

Reply:
This will break OSS users existing implementation of custom metrics reporter I think? We need to call it out in the release notes if that's the case.

Reply:
Yes. We need to mention in the release note in 1.17

Reply:
currently, it is not possible to implement custom metrics reporter. because the people don't have access to internal metrics definition which is needed to implement that metrics.Client which takes a scopeId (int), and the implementation need to translate to string but cannot access those internal definition.

Reply:
we need to fix that as well.